File: AuthService/src/main/java/com/slavacom/authservice/event/UserLoginEvent.java
package com.slavacom.authservice.event;

import java.util.UUID;

public record UserLoginEvent(UUID eventId, UUID userId, String email, String type) {
	public UserLoginEvent(UUID eventId, UUID userId, String email) {
		this(eventId, userId, email, "USER_LOGIN_EVENT");
	}
}

File: NotificationService/src/main/kotlin/com/slavacom/notificationservice/event/user/UserLoginEvent.kt
package com.slavacom.notificationservice.event.user;

import java.util.*

@JvmRecord
data class UserLoginEvent(val eventId: UUID?, val userId: UUID?, val email: String?, val type: String?) {
    constructor(eventId: UUID?, userId: UUID?, email: String?) : this(eventId, userId, email, "USER_LOGIN_EVENT")
}